src/Apps/Artwork/Components/ArtworkBanner/index.tsx

```diff
@@ -0,0 +1,154 @@
+import { renderWithLoadProgress } from "Artsy/Relay/renderWithLoadProgress"
+import { ContextConsumer } from "Artsy/Router"
+import React from "react"
+import { createFragmentContainer, graphql, QueryRenderer } from "react-relay"
+import { get } from "Utils/get"
+import { Banner } from "./Banner"
+
+import { ArtworkBanner_artwork } from "__generated__/ArtworkBanner_artwork.graphql"
+import { ArtworkBannerQuery } from "__generated__/ArtworkBannerQuery.graphql"
+
+export interface ArtworkBannerProps {
+  artwork: ArtworkBanner_artwork
+}
+
+export class ArtworkBanner extends React.Component<ArtworkBannerProps> {
+  render() {
+    const { context, partner } = this.props.artwork
+    if (!context) return null
+    // imageUrl: image for avatar
+    // initials: fallback partner initials in case image is not there.
+    // meta: in auction / at fair / in show
+    // name:  auction / fair / show name
+    // subHeadline: partner name
+    switch (context.__typename) {
+      case "ArtworkContextAuction":
+        const auctionImage = get(partner, p => p.profile.icon.url)
+        return (
+          <Banner
+            imageUrl={auctionImage}
+            initials={partner.initials}
+            meta="In auction"
+            name={context.name}
+            subHeadline={partner.name}
+          />
+        )
+      case "ArtworkContextFair":
+        const fairImage = get(context, c => c.profile.icon.img.url)
+        const initials = get(context, c => c.profile.initials)
+        return (
+          <Banner
+            imageUrl={fairImage}
+            initials={initials}
+            meta="At fair"
+            name="Fair name goes here" // {context.name}
+            subHeadline={partner.name}
+          />
+        )
+      case "ArtworkContextPartnerShow":
+        const showImage = get(context, c => c.thumbnail.img.url)
+        let showLine = "In current show"
+        if (context.status === "upcoming") {
+          showLine = "In upcoming show"
+        } else if (context.status === "closed") {
+          showLine = "In past show"
+        }
+        return (
+          <Banner
+            imageUrl={showImage}
+            initials={partner.initials}
+            meta={showLine}
+            name="Show name goes here" // {context.name}
+            subHeadline={partner.name}
+          />
+        )
+      default:
+        return null
+    }
+  }
+}
+
+export const ArtworkBannerFragmentContainer = createFragmentContainer(
+  ArtworkBanner,
+  graphql`
+    fragment ArtworkBanner_artwork on Artwork {
+      context {
+        __typename
+        ... on ArtworkContextAuction {
+          name
+          href
+          is_auction
+          is_closed
+          is_open
+          live_start_at
+          live_url_if_open
+        }
+        ... on ArtworkContextFair {
+          name
```

Comment:
@alloy - wondering if you might have an idea of what is going on here. We are getting context for the artwork in this query that can be one of 3 types: ArtworkContextAuction, ArtworkContextFair or ArtworkContextPartnerShow. Based on what is the type we are requesting proper object in MP (https://github.com/artsy/metaphysics/blob/master/src/schema/artwork/context.js#L44). All the underlying objects have name/href properties. And  running this query on MP directly returns proper data. However defining it here as part of ArtworkBannerFragmentContainer - the name/href is dropped for some reason for Fairs and Shows. It's not included in src/__generated__/ArtworkBanner_artwork.graphql.ts below. Any thoughts?